src/core/ddsc/src/dds_data_allocator.c

```diff
@@ -126,3 +126,30 @@ void dds_data_allocator_free (dds_data_allocator_t *data_allocator, void *ptr)
   ddsrt_free (ptr);
 #endif
 }
+
+bool is_loan_available(const dds_entity_t entity)
+{
+  bool ret = false;
+#ifdef DDS_HAS_SHM
+  dds_entity * e;
+
+  if (DDS_RETCODE_OK != dds_entity_pin(entity, &e)) {
+    return ret;
+  }
+
+  switch (dds_entity_kind(e)) {
+    case DDS_KIND_READER:
+      ret = ((struct dds_reader *) e)->m_entity.m_domain->gv.config.enable_shm;
```

Comment:
Actually, it would be better to do a different check: whether `((struct dds_reader *) e)->->m_iox_sub` is not a null pointer (and the same for the writer). It used to be this test, but @reicheratwork changes to make it dependent on the QoS did change that. Looking at this, I realize there are now a number of places where the change still needs to be made:

* `dds_write_impl` (though it is conceivable that @e-hndrks's PR affects that)
* `dds__reader_data_allocator_init` and `dds__writer_data_allocator_fini`

Reply:
@eboasson Will fix this and other instances